Introduce ExternalLink control

jarolrod opened the pull request:
Control to handle storing and opening of external links. Can be just text or icon, or combination. To be tested with #143

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/141)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/141)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/141)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/141)

shaavan commented on src/qml/controls/ExternalLink.qml:
```diff
@@ -0,0 +1,57 @@
+
+Control {
+  id: root
+  property string link
```
```suggestion
  required property string link
```

One important addition I would suggest is to make the link property required. Since the component is used to create an external link, using this component must imply providing the link itself.

jarolrod replied:
done in [b700fc6](https://github.com/bitcoin-core/gui-qml/commit/b700fc6ba2ca86e047fe5db0c44e3dab4c7694ac)

shaavan reviewed (commented):
Concept ACK

Keeping #143 in mind,  porting the code related to creating an external link to separate files makes sense for the following reasons:
1. This will allows the component to be used separately from the Settings.qml component.
2. This will allow link-related code to be easily maintainable and also more intuitive to use compared to the current implementation.

jarolrod commented:
Updated from 419bd21 to b700fc6 ([pr141.01](https://github.com/jarolrod/gui-qml/tree/pr141.01) -> [pr141.02](https://github.com/jarolrod/gui-qml/tree/pr141.02), [diff](https://github.com/jarolrod/gui-qml/compare/pr141.01..pr141.02))


changes: 
  - make link a required property.
  - remove uncessary and redundant layout formatting declarations.
  - remove wordwrap, this was a part of the implementaion in information.qml but this is not needed at all according to the design specs, if anything a follow-up can add eliding instead. Removing word-wrap eliminates the formatting issue with long url descriptions seen in #143.
  - Refactor the component holding the icon from `Image` to `Button`. 
    - we want the icon to be represented by the `Icon` type because the icon will actually appear sharper on screen, and because we can't colorize an `Image` without `QtGraphicalEffects`. If we added support now for `QtGraphicalEffects` it is endlessly more complex to colorize an `Image` than an `Icon`.
  - edit commit message to reflect that this is introducing a control and not a component.

jarolrod commented:
updated from b700fc6 to 7d1d977 ([pr141.02](https://github.com/jarolrod/gui-qml/tree/pr141.02) -> [pr141.03](https://github.com/jarolrod/gui-qml/tree/pr141.03), [diff](https://github.com/jarolrod/gui-qml/compare/pr141.02..pr141.03))

changes: handle undefined state of `description` and `iconSource` by setting it as an empty string

jarolrod commented:
updated from 7d1d977 to aadee41 ([pr141.03](https://github.com/jarolrod/gui-qml/tree/pr141.03) -> [pr141.04](https://github.com/jarolrod/gui-qml/tree/pr143.04), [diff](https://github.com/jarolrod/gui-qml/compare/pr141.03..pr143.04))

Changes: make spacing in the RowLayout 0, as the design spec specifies no spacing between the link description and the icon box.

hebasto reviewed (approved):
ACK aadee41e0bfdf703b94c9d1e99af57b1c2f7dd32, tested on Ubuntu 22.04.
